core/src/main/java/de/danoeh/antennapod/core/storage/DBTasks.java

```diff
@@ -144,53 +144,37 @@ public static void playMedia(final Context context, final FeedMedia media,
     private static final AtomicBoolean isRefreshing = new AtomicBoolean(false);
 
     /**
-     * Refreshes a given list of Feeds in a separate Thread. This method might ignore subsequent calls if it is still
+     * Refreshes all feeds.
+     * It must not be from the main thread.
+     * This method might ignore subsequent calls if it is still
      * enqueuing Feeds for download from a previous call
      *
      * @param context  Might be used for accessing the database
-     * @param feeds    List of Feeds that should be refreshed.
      */
-    public static void refreshAllFeeds(final Context context, final List<Feed> feeds) {
-        refreshAllFeeds(context, feeds, null);
-    }
-
-    /**
-     * Refreshes a given list of Feeds in a separate Thread. This method might ignore subsequent calls if it is still
-     * enqueuing Feeds for download from a previous call
-     *
-     * @param context  Might be used for accessing the database
-     * @param feeds    List of Feeds that should be refreshed.
-     * @param callback Called after everything was added enqueued for download. Might be null.
-     */
-    public static void refreshAllFeeds(final Context context, final List<Feed> feeds, @Nullable Runnable callback) {
+    public static void refreshAllFeeds(final Context context) {
         if (!isRefreshing.compareAndSet(false, true)) {
             Log.d(TAG, "Ignoring request to refresh all feeds: Refresh lock is locked");
             return;
         }
 
-        new Thread(() -> {
-            if (feeds != null) {
-                refreshFeeds(context, feeds);
-            } else {
-                refreshFeeds(context, DBReader.getFeedList());
-            }
-            isRefreshing.set(false);
+        if (Looper.myLooper() == Looper.getMainLooper()) {
+            Log.wtf(TAG, "DBTasks.refreshAllFeeds() must not be called from the main thread");
```

Comment:
I think it might make sense to crash the app here by throwing an exception. Otherwise, the log message might be overlooked.

Reply:
I thought `Log.wtf()` crashes the app. If not, what is the preferred way to do so?

Reply:
I did not know that. Looks like this is not always the case: https://stackoverflow.com/q/39797350/4193263. I'd prefer to use a normal exception (should be more readable, too).

Reply:
done